### PR TITLE
Add Neighborhood field in BasePostalAddress

### PR DESCRIPTION
**Pull request by aarthi-axim**
Added `Neighborhood` field in `BasePostalAddress`

### CODE REVIEW

File: apps/addresses/models.py [hunk 1]
```diff
@@ -16,19 +16,22 @@ class BasePostalAddress(AbstractGeolocation):
     This object is always referenced by a foreign key from another object
 
     Storing address as a separate model is a best practice
-    Other than that, there are none. Parsing postal addresses and (even human names, for that matter) are notoriously difficult (not even taking international addresses into consideration)
```
Comment (aarthi-axim) [inline apps/addresses/models.py:19]: ## REVIEWER NOTE:
Lint fix

File: apps/addresses/models.py [hunk 2]
```diff
@@ -16,19 +16,22 @@ class BasePostalAddress(AbstractGeolocation):
...
 
     Right now, go with the "good enough" approach
     """
     name = models.CharField(max_length=64, blank=True)
     street = models.CharField(max_length=128, blank=True)
+    neighborhood = models.CharField(max_length=64, blank=True)
     city = models.CharField(max_length=64, blank=True)
     state = models.CharField(max_length=2, blank=True)
-    zipcode = models.CharField(max_length=5, blank=True)
+    zipcode = models.CharField(max_length=16, blank=True)
```
Comment (aarthi-axim) [inline apps/addresses/models.py:34]: ## REVIEWER NOTE:
Changed `max_length` to make it compatible with `zippy model`

File: apps/addresses/models.py [hunk 3]
```diff
@@ -46,7 +49,7 @@ def __str__(self):
         return value
 
     def clone(self):
-        address_clone = PostalAddress.objects.create(
```
Comment (aarthi-axim) [inline apps/addresses/models.py:49]: ## REVIEWER NOTE:
Cloning should be done for self model(`BasePostalAddress`)

File: apps/addresses/models.py [hunk 4]
```diff
@@ -120,7 +123,7 @@ def get_static_google_map_image_url(self):
         height = 234
         str_value = self.__str__()
         base_url = 'http://maps.googleapis.com/maps/api/staticmap?'
-        query = urllib.urlencode(
```
Comment (aarthi-axim) [inline apps/addresses/models.py:123]: ## REVIEWER NOTE:
`urllib` from `six.moves` has no `urlencode` instead it should be imported from `urllib.parse`

File: apps/addresses/models.py (hunk 2 above)
Comment (jontsai) [reply apps/addresses/models.py:34]: Somewhat leaning toward keeping it as is for now, and modifying it in the subclass

File: apps/addresses/models.py [hunk 5]
```diff
@@ -46,7 +49,7 @@ def __str__(self):
         return value
 
     def clone(self):
-        address_clone = PostalAddress.objects.create(
+        address_clone = BasePostalAddress.objects.create(
```
Comment (jontsai) [inline apps/addresses/models.py:52]: This is not correct. I think It would need to be changed to something like `self.__class__.objects.create(...)`

**Review by jontsai — CHANGES_REQUESTED** [review]

File: apps/addresses/models.py (hunk 5 above)
Comment (aarthi-axim) [reply apps/addresses/models.py]: Done

**Review by jontsai — APPROVED** [review]

**Pull request merged**